scipy/stats/tests/test_morestats.py

```diff
@@ -2092,6 +2092,78 @@ def test_overflow(self, x):
         y_extreme = special.boxcox(x_treme, lmbda)
         assert_allclose(y_extreme, ymax * np.sign(lmbda))
 
+    def test_negative_ymax(self):
+        with pytest.raises(ValueError, match="`ymax` must be strictly positive"):
+            stats.boxcox_normmax(self.x, ymax=-1)
+
+    @pytest.mark.parametrize("x", [
+        # positive overflow in float64
+        np.array([2003.0, 1950.0, 1997.0, 2000.0, 2009.0],
+                 dtype=np.float64),
+        # negative overflow in float64
+        np.array([0.50000471, 0.50004979, 0.50005902, 0.50009312, 0.50001632],
+                 dtype=np.float64),
+    ])
+    @pytest.mark.parametrize("ymax", [1e10, 1e100, None])
+    # TODO: add method "pearsonr" after fix overflow issue
+    @pytest.mark.parametrize("method", ["mle"])
+    def test_user_defined_ymax_input_float64(self, x, ymax, method):
+        # Test the maximum of the transformed data close to ymax
+        with pytest.warns(UserWarning, match="The optimal lambda is"):
+            kwarg = {'ymax': ymax} if ymax is not None else {}
+            lmb = stats.boxcox_normmax(x, method=method, **kwarg)
+            x_treme = [np.min(x), np.max(x)]
+            ymax_res = max(abs(stats.boxcox(x_treme, lmb)))
+            if ymax is None:
+                # 10000 is safety factor used in boxcox_normmax
+                ymax = np.finfo(x.dtype).max / 10000
+            assert_allclose(ymax, ymax_res, rtol=1e-1)
```

Comment:
IIRC, this tolerance needed to be loose before because I had suggested the wrong branch of `lambertw`. With the branch corrected, can this be tighter?

Reply:
I've updated the tolerance to 1e-5.